lib/internal/child_process.js

```diff
@@ -379,6 +379,9 @@ ChildProcess.prototype.spawn = function(options) {
 
       if (i > 0 && this.pid !== 0) {
         this._closesNeeded++;
+        if (stream._events && stream._events.close) {
+          stream.socket.addListener('close', stream._events.close);
+        }
```

Comment:
I think this would lead to the event handler being called with a wrong `this` value, and I think this might not work if there is more than one listener (because `stream._events.close` would be an array then).

Could we use the listener in the lines below, and call `stream.emit('close')` from there?

Reply:
I see. It might not work when events is an array.

I want to fix.
`createSocket` create new Socket object and sharing only handle of old socket.
Therefore, we can't call old close events.
How do i fix it best ?

https://github.com/nodejs/node/blob/5a3e445304f3920590b2ae1d1237fe7615572a44/lib/internal/child_process.js#L274
https://github.com/nodejs/node/blob/5a3e445304f3920590b2ae1d1237fe7615572a44/lib/internal/child_process.js#L377

Reply:
> Therefore, we can't call old close events.

I’m not sure I understand this – Can you explain more? i.e. what wouldn’t work if we add `stream.emit('close');` in the callback below?

Reply:
I was misunderstanding. It could use `stream.emit('close')`, if new stream object have old stream object instead of having `_events`.

I am going to fix.
```
if (stream._stdio) {
  stream._stdio.emit('close');
}
```

```
  handle: handle,
- _events: events 
+ _stdio: stdio
```